HIPs/firefly.md

```diff
@@ -0,0 +1,374 @@
+
+While there are reasons for this slow time horizon that span technical, business, and governance, FireFly aims to eliminate one of the most significant - the “plumbing problem”. At a high level, Enterprise blockchain use cases attempt to coordinate data flows and transactions across disparate, separately owned systems (ERP, mainframe, commerce, etc.). To do this end-to-end, four things are required:
+
+1. the _single source of truth coordination system_ across parties
+2. _per-member_ _plumbing software_ to enable reliable, private integration to and from the coordination system
+3. _cross-org plumbing software_ that provides robust cross-connectivity among members alongside (1) for use when the coordination system is not suitable or not preferred as the data flow mechanism. Examples include sensitive data, low latency messaging, and large data flows
+4. _a control plane_ to manage all of these runtime and configuration details
+
+We define a **multi-party system** as the set of technology, infrastructure, and management capability to enable a consortium to build and deploy a use case which includes (1), (2), (3), and (4). We observe nearly all the real world consortia that we worked with over the last six years require (1), (2), (3), and (4) to build and deploy an enterprise blockchain use case. In our experience, consortia spend a huge amount of their resources building out the plumbing portion of the system. Furthermore, it is our observation that companies spend most of their budget on (2) and (3).
+
+Hyperledger FireFly brings together (1), (2), (3), and (4) to offer a pre-integrated set of runtimes via a highly pluggable framework. Note that FireFly is NOT attempting to replace or invalidate any specific runtime technology, but rather pulls together multiple complementary technologies into a coherent, larger system. Indeed, many of the infrastructure runtimes FireFly relies on are separate open source projects.
+
+Enterprise blockchains are purpose built to deliver (1) with invaluable attributes like _global ordering_ and _finality_. FireFly will support Hyperledger Fabric, Enterprise Ethereum (such as Quorum and Besu), and will investigate Corda support. (1) also includes global storage providers such as IPFS. FireFly also provides a significant amount of technology for (2) and (3) including cross-party messaging, cross-party private document transfer, and event propagation. All of these runtimes and connectors into the FireFly node. For (4), Hyperledger FireFly also provides a network map and registry that unifies all of the identity and bindings information needed.
+
+## Motivation
+
+When building a system for multiparty business transactions based on blockchain technology, there are a number of problems that need to be solved. Problems such as:
+
+- How do parties agree on standardized data types?
```

Comment:
- How do parties agree on standardized data types?
this is indeed problems any multi-party system face, regardless of whether they are built with blockchain or not, it's listed here because blockchain plays a key role in the solution, utilizing the atomic broadcast feature of a blockchain.

- How do I securely send private data between two parties in the system without other parties being able to view the data?
this is listed here because of the need for a private transaction solution that can accommodate various data types, including especially those not suitable for the built-in privacy solutions that use the state DB for data storage (large files in particular)

Reply:
I've added some clarification below the bullet list based on the discussions so far.